WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalManagerBase.java

```diff
@@ -73,4 +75,21 @@ public IntervalBase getIntervalOfType(IntervalType type) {
 		}
 		return collectedIntervals;
 	}
+
+	/**
+	 * @return An {@link ArrayList} of intervals of the specified Perspective
+	 *         type.
+	 */
+	protected List<IntervalBase> getPerspectiveIntervalsOfType(Perspective type) {
```

Comment:
This method is more similar to `getEditorIntervalsOfDocType(DocumentType type)` than to `getIntervalsOfType`, since it is fetching intervals of specific Perspective type (with `getIntervalsOfType`, it would be possible just to get all Perspective intervals). Furthermore, with `getIntervalsOfType` method, it would need 2 iterations over list to fetch perspective-specific intervals, while this method requires only 1 iteration. Is that enough to keep this method?